### PR TITLE
add mergebamalignment

### PR DESCRIPTION
**Pull request by KevinMenden**
Adding `gatk MergeBamAlignment` 

https://github.com/nf-core/modules/issues/227

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/gatk4/mergebamalignment/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    conda (params.enable_conda ? "bioconda::gatk4=4.2.0.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/gatk4:4.2.0.0--0"
+    } else {
+        container "quay.io/biocontainers/gatk4:4.2.0.0--0"
+    }
+
+    input:
+    tuple val(meta), path(aligned)
+    path unmapped
```
Comment (drpatelh) [inline software/gatk4/mergebamalignment/main.nf:23]:
```suggestion
    path unaligned
```

Comment (drpatelh) [reply software/gatk4/mergebamalignment/main.nf:23]: Maybe go with either `aligner` -> `unaligned` or `mapped` -> `unmapped`?

File: software/gatk4/mergebamalignment/main.nf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    path fasta
+    path dict
+
+    output:
+    tuple val(meta), path('*.merged.bam')       , emit: bam
+    path  '*.version.txt'                       , emit: version
```
Comment (drpatelh) [inline software/gatk4/mergebamalignment/main.nf:29]:
```suggestion
    tuple val(meta), path('*.merged.bam'), emit: bam
    path  '*.version.txt'                , emit: version
```

Comment (drpatelh) [reply software/gatk4/mergebamalignment/main.nf:29]: Aaannnndd he's back! 😅

File: tests/data/README.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ When adding a new module, please check carefully whether the data necessary for
         * 'test_{,methylated}_paired_end.sorted.bam': sorted version of the above bam file
         * 'test_{,methylated}_paired_end.bam.sorted.bam.bai': bam index for the sorted bam file
         * 'test_single_end.bam': alignment (unsorted) of the 'test_1.fastq.gz' reads against test_genomic.fasta using minimap2
+        * 'test_unaligned.bam': unmapped BAM file create from 'test_1.fastq.gz' using GATK4 SamToFastq
```
Comment (drpatelh) [inline tests/data/README.md:17]:
```suggestion
        * 'test_unaligned.bam': unmapped BAM file created from 'test_1.fastq.gz' using GATK4 SamToFastq
```

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looks great @KevinMenden ! Couple of minor gripes.

File: software/gatk4/mergebamalignment/main.nf (hunk 2 above)
Comment (KevinMenden) [reply software/gatk4/mergebamalignment/main.nf:29]: 😁 

it's never gonna happen!!

File: software/gatk4/mergebamalignment/main.nf (hunk 1 above)
Comment (KevinMenden) [reply software/gatk4/mergebamalignment/main.nf:23]: 👍

File: software/gatk4/mergebamalignment/meta.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test']
+  - aligned:
+      type: file
+      description: The aligned bam file
+      pattern: "*.{bam}"
+  - unmapped:
```
Comment (drpatelh) [inline software/gatk4/mergebamalignment/meta.yml:46]:
```suggestion
  - unaligned:
```

**Review by drpatelh — APPROVED** [review]
Need to remove the updates to `.github/filters.yml` on the PR and add to `tests/config/pytest_software.yml` instead.

**Pull request merged**